### PR TITLE
Saas api key

### PR DESCRIPTION
**Pull request by pjagielski**

### CODE REVIEW

File: src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java [hunk 1]
```diff
@@ -59,4 +63,8 @@ public String consumeAndLogEntity(@NotNull CloseableHttpResponse response) throw
         log.info("Entity {}: {}", REQUEST_COUNTER, content);
         return content;
     }
+
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Name 'REQUEST_COUNTER' must match pattern '^[a-z][a-zA-Z0-9]*$'.

Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Variable 'REQUEST_COUNTER' explicitly initialized to '0' (default value for its type).

File: src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java [hunk 2]
```diff
@@ -23,9 +23,11 @@
     private static Integer SOME_PULL_REQUEST_ID = 12314;
     private static String SOME_REPOSITORY = "repo";
     private static String SOME_PROJECT = "proj";
+    private static String SOME_API_KEY = "my_api_key";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java]: [Checkstyle] ERROR: Name 'SOME_API_KEY' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java (hunk 1 above)
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Name 'REQUEST_COUNTER' must match pattern '^[a-z][a-zA-Z0-9]*$'.

Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Variable 'REQUEST_COUNTER' explicitly initialized to '0' (default value for its type).

File: src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java [hunk 3]
```diff
@@ -20,12 +20,14 @@
 
 public class SaasFacadeTest extends HttpConnectorEnv {
 
-    private static Integer SOME_PULL_REQUEST_ID = 12314;
-    private static String SOME_REPOSITORY = "repo";
-    private static String SOME_PROJECT = "proj";
+    private final static Integer SOME_PULL_REQUEST_ID = 12314;
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java]: [Checkstyle] ERROR: 'static' modifier out of order with the JLS suggestions.

File: src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java [hunk 4]
```diff
@@ -20,12 +20,14 @@
...
+    private final static String SOME_REPOSITORY = "repo";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java]: [Checkstyle] ERROR: 'static' modifier out of order with the JLS suggestions.

File: src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java [hunk 5]
```diff
@@ -20,12 +20,14 @@
...
+    private final static String SOME_PROJECT = "proj";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java]: [Checkstyle] ERROR: 'static' modifier out of order with the JLS suggestions.

File: src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java [hunk 6]
```diff
@@ -20,12 +20,14 @@
...
+    private final static String SOME_API_KEY = "my_api_key";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/saas/SaasFacadeTest.java]: [Checkstyle] ERROR: 'static' modifier out of order with the JLS suggestions.

File: src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java (hunk 1 above)
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Name 'REQUEST_COUNTER' must match pattern '^[a-z][a-zA-Z0-9]*$'.

Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/http/HttpConnector.java]: [Checkstyle] ERROR: Variable 'REQUEST_COUNTER' explicitly initialized to '0' (default value for its type).

File: src/main/java/pl/touk/sputnik/configuration/CliOption.java [hunk 7]
```diff
@@ -9,7 +9,8 @@
     CONF("cli.conf", "Configuration properties file", null),
     CHANGE_ID("cli.changeId", "Gerrit change id", null),
     REVISION_ID("cli.revisionId", "Gerrit revision id", null),
-    PULL_REQUEST_ID("cli.pullRequestId", "Stash pull request id", null);
+    PULL_REQUEST_ID("cli.pullRequestId", "Stash pull request id", null),
+    API_KEY("cli.apiKey", "Optional API key", null);
```
Comment (SpOOnman) [inline src/main/java/pl/touk/sputnik/configuration/CliOption.java]: I miss an information in description: what is this API key intended for?

**Comment by SpOOnman** [comment]
Only one comment on my side.

**Comment by pjagielski** [comment]
Documented apiKey parameter

**Pull request merged**